sql/moz-fx-data-shared-prod/subscription_platform_derived/stripe_logical_subscriptions_history_v1/query.sql

```diff
@@ -62,14 +62,19 @@ subscriptions_history_charge_summaries AS (
   SELECT
     history.id AS subscriptions_history_id,
     ARRAY_AGG(
-      cards.country IGNORE NULLS
+      IF(
+        cards.country IS NOT NULL,
+        STRUCT(
+          cards.country AS latest_card_country,
+          COALESCE(charge_us_zip.state_code, charge_ca_post.province_code) AS latest_card_state
+        ),
+        NULL
+      ) IGNORE NULLS
       ORDER BY
-        -- Prefer charges that succeeded.
+        -- Prefer charges that succeeded
         IF(charges.status = 'succeeded', 1, 2),
         charges.created DESC
-      LIMIT
-        1
```

Comment:
I'd recommend leaving `LIMIT 1` in.  While it isn't technically required, I believe it does help a bit with performance since BigQuery can just keep a single record per aggregation rather than aggregating them all and only using one.